dvci/shc-web-verifier
# Cordova iOS Styling Fixes

## Background and Motivation
There is no WebRTC for iOS, so we are using the [cordova-plugin-iosrtc](https://github.com/cordova-rtc/cordova-plugin-iosrtc) plugin to expose the WebRTC W3C API for our Cordova iOS app.

This plugin places a native UIView layer on top of the HTML video that we use in the `QrScan` component. Therefore, some of the video properties are artificially set by the plugin because there is no “real” video attached to the `<video>` element. While most of the styling properties remain in tact after using this plugin, some of them are not applied, making the iOS app inconsistent with the browser and Cordova Android app.

## Behavior changes
The following issues aree resolved in this PR:
* **Z-index styling with the image frame and video element:** the UIView layer over the video prevents the current z-index from being respected, and so the scan frame appeared behind the video. As suggested by the Github issues for the plugin, the `qrscanner` z-index was set to -1, and the document body was set to transparent so that the UIView video can still be seen.
* **Disable zoom:** When toggling between portrait and landscape mode, the screen zooms in and it becomes difficult to return to the original zoom size. The meta viewport was updating to disallow zoom.
* **Disable over scroll in portrait mode:** In portrait mode, the user can scroll way below the scanner frame, and the video will no longer align up with the frame. To prevent this, the config.xml was updating to disallow over scroll.
* **Disable over scroll in landscape mode:** After reading through various [Github issues](https://github.com/cordova-rtc/cordova-plugin-iosrtc/issues/478#issuecomment-600001710), it seems that the intended behavior is to not refresh the video automatically when scrolling. The “disallow over scroll” preference only applies to portrait mode, and so a workaround was implemented based on the Github issue response from the plugin’s creator.
* **Allow for camera switching:** The camera switch button did not work on iOS because the stream would not update even after detecting the new camera id. The dependency array was updated to include the `cameraDeviceId`, which allowed the stream to properly refresh.

## Testing Guidance
It’s important that we test this branch on iOS, Android, and in the browser to make sure that nothing breaks from these updates.

iOS:
* Read through this [iOS testing Confluence document](https://confluence.mitre.org/pages/viewpage.action?pageId=294293841) to set up Xcode for testing.
* Try testing on a physical device and on a simulated device that is a different model from the physical device (ex. Try testing on a physical iPhone and simulated iPad to see how the styling looks for different dimensions)

Android:
* Try testing on a physical Android device to see that the camera’s functionality has not been affected by the dependency array updates in the `QrScan` component (I wasn’t able to test this on a simulated device)
* Check that the styling looks correct

Browser:
* Try testing the camera’s functionality — I noticed when testing that the camera is slower now that the `cameraDeviceId` is included in the dependency array (going to look into this in a separate PR that collectively addresses the camera console errors)
* Check that the styling looks correct
